File: leo2-common/src/test/java/org/deku/leo2/MessagingTest.java
package org.deku.leo2;

import ch.qos.logback.classic.Level;
import ch.qos.logback.classic.Logger;
import org.junit.After;
import org.junit.Before;
import sx.jms.embedded.Broker;
import sx.jms.embedded.activemq.ActiveMQBroker;

/**
 * Created by masc on 16.06.15.
 */
public abstract class MessagingTest {
    private String mHttpUrl = "http://localhost:8080/leo2/jms";
    //private String mNativeUrl = "tcp://localhost:61616";
    private String mNativeUrl = "vm://localhost?create=false";

    private Broker mBroker;

    @Before
    public void setup() throws Exception {
        // Log levels
        Logger root = (Logger) org.slf4j.LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
        root.setLevel(Level.INFO);
        Logger lMessaging = (Logger)org.slf4j.LoggerFactory.getLogger("org.deku.leo2.messaging");
        lMessaging.setLevel(Level.TRACE);

        // Start broker
        mBroker = ActiveMQBroker.instance();
        mBroker.start();
    }

    @After
    public void tearDown() throws Exception {
        mBroker.stop();
    }
}

File: libs/leoz-common/src/test/java/org/deku/leoz/MessagingTest.kt
package org.deku.leoz

import ch.qos.logback.classic.Level
import ch.qos.logback.classic.Logger
import org.deku.leoz.config.ActiveMQConfiguration
import org.junit.After
import org.junit.Before
import sx.jms.Broker
import sx.jms.activemq.ActiveMQBroker

/**
 * Created by masc on 16.06.15.
 */
abstract class MessagingTest {
    /**
     * Test broker
     */
    val broker by lazy {
        ActiveMQConfiguration.instance.broker
    }

    @Before
    @Throws(Exception::class)
    fun setup() {
        this.broker.start()
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        this.broker.stop()
    }
}
